src/transport_ofi.c

```diff
@@ -542,57 +542,33 @@ void init_bounce_buffer(shmem_free_list_item_t *item)
 
 
 static inline
-int bind_enable_cq_ep_resources(shmem_transport_ctx_t *ctx)
+int bind_enable_ep_resources(shmem_transport_ctx_t *ctx)
 {
     int ret = 0;
 
     /* Attach the shared context */
-    ret = fi_ep_bind(ctx->cq_ep, &shmem_transport_ofi_stx_pool[ctx->stx_idx].stx->fid, 0);
-    OFI_CHECK_RETURN_STR(ret, "fi_ep_bind STX to CQ endpoint failed");
-
-    /* Attach CQ for obtaining completions for buffered puts */
-    ret = fi_ep_bind(ctx->cq_ep, &ctx->cq->fid, FI_SEND);
-    OFI_CHECK_RETURN_STR(ret, "fi_ep_bind CQ to CQ endpoint failed");
-
-    /* Attach the address vector */
-    ret = fi_ep_bind(ctx->cq_ep, &shmem_transport_ofi_avfd->fid, 0);
-    OFI_CHECK_RETURN_STR(ret, "fi_ep_bind AV to CQ endpoint failed");
-
-    ret = fi_enable(ctx->cq_ep);
-    OFI_CHECK_RETURN_STR(ret, "fi_enable on CQ endpoint failed");
-
-    return ret;
-}
-
-
-static inline
-int bind_enable_cntr_ep_resources(shmem_transport_ctx_t *ctx)
-{
-    int ret = 0;
-
-    /* Attach the shared context */
-    ret = fi_ep_bind(ctx->cntr_ep, &shmem_transport_ofi_stx_pool[ctx->stx_idx].stx->fid, 0);
-    OFI_CHECK_RETURN_STR(ret, "fi_ep_bind STX to CNTR endpoint failed");
+    ret = fi_ep_bind(ctx->ep, &shmem_transport_ofi_stx_pool[ctx->stx_idx].stx->fid, 0);
+    OFI_CHECK_RETURN_STR(ret, "fi_ep_bind STX to endpoint failed");
```

Comment:
I think I understand what's happening.  We no longer have a CQ with the FI_SEND flag, but "selective completion" allows us to specifying FI_COMPLETION on writemsg() operations to increment that completion counter.  And you need the increment the pending_put counter now, whereas before we didn't care about that on the CQ EP, right?

I don't see any problems yet... just making sure I understand.  The overall merge actually appears pretty straight forward.

Reply:
I think the question is: this PR changes the the CQ to EP binding flags from `FI_SEND` to `FI_SELECTIVE_COMPLETION | FI_TRANSMIT`.  Is this ok?  I believe the answer is yes, `FI_TRANSMIT` encompasses `FI_SEND`, but there might be some subtlety I'm missing.  @sayantansur Can you fact check?

Reply:
You're correct, Jim.

Reply:
Yeah, sorry for my poor wording above - I think I was asking 2 questions.  That's exactly my 1st question about removing `FI_SEND`.

My 2nd question was why we increment the pending_put counter for communications that were originally on the `cq_eq` only.  I suppose we need to do that now that it's on the merged EP where we must track all pending ops in quiet(s).

Reply:
Ah yeah, thanks.